Library/Formula/aqbanking.rb

```diff
@@ -31,17 +20,60 @@ def install
                           "--prefix=#{prefix}",
                           "--enable-cli",
                           "--with-gwen-dir=#{HOMEBREW_PREFIX}"
-    system "make install"
+    system "make", "install"
   end
 
-  def caveats; <<-EOS.undent
-    To build the devel version install all the dependencies first,
-    then install the devel version of gwenhywfar separately,
-    and install the devel version of aqbanking afterwards.
+  test do
+    context =<<-EOS.undent
+    accountInfoList {
+      accountInfo {
+        char bankCode="110000000"
+        char bankName="STRIPE TEST BANK"
+        char accountNumber="000123456789"
+        char accountName="demand deposit"
+        char iban="US44110000000000123456789"
+        char bic="BYLADEM1001"
+        char owner="JOHN DOE"
+        char currency="USD"
+        int  accountType="0"
+        int  accountId="2"
+
+        statusList {
+          status {
+            int  time="1388664000"
+
+            notedBalance {
+              value {
+                char value="123456%2F100"
+                char currency="USD"
+              } #value
+
+              int  time="1388664000"
+            } #notedBalance
+          } #status
+
+          status {
+            int  time="1388750400"
+
+            notedBalance {
+              value {
+                char value="132436%2F100"
+                char currency="USD"
+              } #value
+
+              int  time="1388750400"
+            } #notedBalance
+          } #status
+        } #statusList
 
-    brew install aqbanking --devel --only-dependencies
-    brew install gwenhywfar --devel
-    brew install aqbanking --devel
+      } # accountInfo
+    } # accountInfoList
     EOS
+    require "open3"
+    Open3.popen3("#{bin}/aqbanking-cli", "listbal") do |stdin, stdout, _|
```

Comment:
I just saw that in the [libtiff](https://github.com/Homebrew/homebrew/blob/9f6926265f8e4be7cc80dfe9042f2cd3c1e8dc9e/Library/Formula/libtiff.rb) formula, that it is also possible to write testfiles.

Which approach is preferred by Homebrew?
The `popen3` one here of the `(testfiles/test).write` one similar to the on in libtiff?